README.md

```diff
@@ -1,191 +1,323 @@
+    *   Skip AWS setup and agent installation
+    *   Choose whether to receive email status reports for the project
+
+## Set Up A Dev Environment
+
+We recommend using [Cloud Shell](https://cloud.google.com/shell/) to prepare
+ts-bridge for deployment to ensure a consistent and stable working environment.
+If you need a dev environment that you can share among multiple users, consider
+using a git repository and
+[open-in-cloud-shell links](https://cloud.google.com/shell/docs/open-in-cloud-shell).
+
+1.  If you are **not** using Cloud Shell:
+    *   Install [go](https://golang.org/)
+    *   Download and install the
+        [Cloud SDK for Go](https://cloud.google.com/sdk/docs/)
+        *   Initialize with the following commands to set the linked project and
+            auth cookie:
+        *   `gcloud init`
+        *   `gcloud auth application-default login`
+    *   Download and install the
+        [original App Engine SDK for Go](https://cloud.google.com/appengine/docs/standard/go/download)
+        (Expand the "Previous App Engine SDK" header to see it)
+1.  Clone the ts-bridge source
+    *   `go get github.com/google/ts-bridge/...`
+    *   The ts-bridge source code should appear in
+        ~/gopath/src/github.com/google/ts-bridge/
+
+## End To End Test (Dev Server)
+
+1.  Ensure that you either have **Owner** permissions for the whole Cloud
+    project, or at minimum the **Monitoring Editor** role
+1.  If using a service account to authenticate, set the
+    [Application Default Credentials](https://cloud.google.com/docs/authentication/production)
+    environment variable to your auth key file
+    *   `export GOOGLE_APPLICATION_CREDENTIALS=/path/to/file.json`
+1.  Create a ts-bridge config with no metrics
+    *   `cd ~/gopath/src/github.com/google/ts-bridge/app; cp metrics.yaml.example metrics.yaml`
+    *   Edit the yaml file, remove the datadog\_metrics sample content, and copy
+        in the name of the project you just created into the
+        stackdriver\_destinations section.
+    *   Your `metrics.yaml` file should look like this:
+    ```
+    datadog_metrics:
+    stackdriver_destinations:
+      - name: stackdriver
+        project_id: "your_project_name"
+    ```
+1.  Turn on the status page (uncomment #ENABLE\_STATUS\_PAGE: "yes" in `app.yaml`)
+1.  Launch a dev server
+    *   `dev_appserver.py app.yaml --port 18080`
+1.  Test via localhost/sync
+    *   `curl http://localhost:18080/sync`
+1.  Verify that no error messages are shown. Troubleshooting guide:
+
+    | Error message | Remedy |
+    | --- | --- |
+    | ERROR: StatsCollector: rpc error: code = PermissionDenied desc = The caller does not have permission | Ensure the authenticating user has at least the "Monitoring Editor" role |
+
+1.  Configure metrics by getting your
+    [API and application keys](https://docs.datadoghq.com/api/?lang=go#overview)
+    from datadog and copying over your sli queries into `metrics.yaml`
```

Comment:
I think mentioning "sli queries" here (and below) might be confusing, since ts-bridge does not really care if you're importing SLI metrics or not. I'd suggest to just say "Datadog queries" instead.

Reply:
done